### PR TITLE
Do not clear fields until request return success in users admin page on user creation

### LINKED ISSUE #13500: [User Management] Info in textbox Username is deleted when email address is wrong.
### Steps to reproduce

1. Go to users
2. Activate send to email address to new users.
3. Write down user/password and a wrong email.
4. Click create new user.

### Expected behaviour
Username  should keep the info due to it was right.

### Actual behaviour

Username textbox is empty after trying to create a new user with a wrong email address.

### Server configuration
**Operating system**:
Ubuntu 14.04 LTS

**Web server:**
Apache

**Database:**
MySQL

**PHP version:**
5.5.9

**ownCloud version:** (see ownCloud admin page)
{"installed":true,"maintenance":false,"version":"8.0.0.1","versionstring":"8.0 alpha 1","edition":""}

**Updated from an older ownCloud or fresh install:**
fresh

**List of activated apps:**
none

**The content of config/config.php:**

```
Insert your config.php content here
(Without the database password and passwordsalt)
```

**Are you using external storage, if yes which one:** local/smb/sftp/...
no

**Are you using encryption:** no
captura de pantalla 2015-01-20 a las 11 15 32

### Client configuration
**Browser:**
Chrome, Firefox

### PR DESCRIPTION
**Pull request by Sugaroverdose**
fixes #13500

### CODE REVIEW

**Comment by MorrisJobke** [comment]
@Sugaroverdose I guess you should sign the contributor agreement and send it to @karlitschek ;)

**Comment by Sugaroverdose** [comment]
Can't i just say MIT? :)

**Comment by MorrisJobke** [comment]
@Sugaroverdose Sure :)

File: settings/js/users/users.js [hunk 1]
```diff
@@ -801,7 +800,7 @@ $(document).ready(function () {
 				GroupList.incEveryoneCount();
 			}).fail(function(result, textStatus, errorThrown) {
 				OC.dialogs.alert(result.responseJSON.message, t('settings', 'Error creating user'));
-			});
+			}).success($('#newuser').get(0).reset());
```
Comment (MorrisJobke) [inline settings/js/users/users.js]: you should wrap it in a `function(){}` ;) Then it works

**Comment by MorrisJobke** [comment]
Works :+1:

**Comment by MorrisJobke** [comment]
cc @LukasReschke @blizzz

**Comment by MorrisJobke** [comment]
@DeepDiver1975 I would like to have this in 8.0. Minor fix, but it helps a lot if you mistyped a name and doesn't have to type in the values again.

**Comment by LukasReschke** [comment]
:+1:

**Comment by MorrisJobke** [comment]
@owncloud-bot this is okay to test

**Comment by Sugaroverdose** [comment]
@MorrisJobke looks like, he didn't heard you :D

**Comment by MorrisJobke** [comment]
@owncloud-bot retest this please

**Comment by MorrisJobke** [comment]
JS Unit test just run fine locally.

**Comment by DeepDiver1975** [comment]
@owncloud-bot retest this please

**Comment by DeepDiver1975** [comment]
@MorrisJobke are there js unit tests on user management?

**Comment by MorrisJobke** [comment]
> @MorrisJobke are there js unit tests on user management?

Yes but just for the delete handler.

**Comment by DeepDiver1975** [comment]
> Yes but just for the delete handler.

too bad ....

**Comment by DeepDiver1975** [comment]
tested :+1:

**Comment by davitol** [comment]
I think that maybe is not well solved.

**Comment by MorrisJobke** [comment]
> I think that maybe is not well solved.

Yes?

**Comment by Sugaroverdose** [comment]
@MorrisJobke if it's needed, i can add regex parsing for email field, but i don't think, that this comment really about this issue
@davitol in both variants email is written not property, why in first case fileds should be erased and in second filed - not?)

**Comment by davitol** [comment]
@Sugaroverdose I think it should be erased in both cases, I wrote down the current behaviour.

**Comment by Sugaroverdose** [comment]
@davitol why it should be erased? if you made typo, or missed one char, isn't it better to leave filed untouched? It's totally radical to clear filed, it's like to say: "hey, you've made a mistake, try to guess where it was"

**Comment by DeepDiver1975** [comment]
from my pov clearing the fields only on success if perfectly fine

**Comment by MorrisJobke** [comment]
I also think that it would be annoying if it gets erased just because it's wrong. Because that is the actual cause for change. You want to modify the wrong version and not have it simply erased.

**Comment by davitol** [comment]
I see your point, but at least maybe it will be necessary to mark which field is wrong.

**Comment by MorrisJobke** [comment]
@davitol Out of scope for this PR. Can you open a issue for this? Thanks

**Comment by Sugaroverdose** [comment]
@MorrisJobke there is already an issue with same ideas

**Comment by MorrisJobke** [comment]
https://github.com/owncloud/core/issues/13503

**Comment by MorrisJobke** [comment]
Ah now I understand you. It's the inline inpute and not the email input for create a user. Lets address this in 8.1

**Comment by jnfrmarks** [comment]
@MorrisJobke 

Why wait for 8.1 to fix the in line issue?

**Comment by MorrisJobke** [comment]
@jnfrmarks I can have a look at this once I have time. There are more critical issues I need to have a look, than a temporarly wrong info. ;)

**Pull request merged**